tests/system/python/packages/test_gcp_gateway.py

```diff
@@ -0,0 +1,134 @@
+CERTS_DIR = "{}/gcp".format(SCRIPTS_DIR_ROOT)
+FOGLAMP_CERTS_DIR = "{}/data/etc/certs/".format(FOGLAMP_ROOT)
+
+
+@pytest.fixture
+def reset_foglamp(wait_time):
+    try:
+        subprocess.run(["cd {}/tests/system/python/scripts/package && ./reset"
+                       .format(PROJECT_ROOT)], shell=True, check=True)
+    except subprocess.CalledProcessError:
+        assert False, "reset package script failed!"
+
+    # Wait for foglamp server to start
+    time.sleep(wait_time)
+
+
+@pytest.fixture
+def remove_and_add_foglamp_pkgs():
+    try:
+        subprocess.run(["cd {}/tests/system/python/scripts/package && ./remove"
+                       .format(PROJECT_ROOT)], shell=True, check=True)
+    except subprocess.CalledProcessError:
+        assert False, "remove package script failed!"
+
+    try:
+        subprocess.run(["cd {}/tests/system/python/scripts/package/ && ./setup {}"
+                       .format(PROJECT_ROOT, package_build_version)], shell=True, check=True)
+    except subprocess.CalledProcessError:
+        assert False, "setup package script failed"
+
+    try:
+        subprocess.run(["sudo apt install -y foglamp-north-gcp-gateway foglamp-south-sinusoid"], shell=True, check=True)
+    except subprocess.CalledProcessError:
+        assert False, "installation of gcp-gateway and sinusoid packages failed"
+
+
+def get_ping_status(foglamp_url):
+    _connection = http.client.HTTPConnection(foglamp_url)
+    _connection.request("GET", '/foglamp/ping')
+    r = _connection.getresponse()
+    assert 200 == r.status
+    r = r.read().decode()
+    jdoc = json.loads(r)
+    return jdoc
+
+
+def get_statistics_map(foglamp_url):
+    _connection = http.client.HTTPConnection(foglamp_url)
+    _connection.request("GET", '/foglamp/statistics')
+    r = _connection.getresponse()
+    assert 200 == r.status
+    r = r.read().decode()
+    jdoc = json.loads(r)
+    return utils.serialize_stats_map(jdoc)
+
+
+def copy_certs():
+    copy_file = "cp {}/* {}".format(CERTS_DIR, FOGLAMP_CERTS_DIR)
+    exit_code = os.system(copy_file)
+    assert 0 == exit_code
+
+
+class TestGCPGateway:
+    def test_gcp_gateway(self, remove_and_add_foglamp_pkgs, reset_foglamp, foglamp_url, wait_time):
+        payload = {"name": "Sine", "type": "south", "plugin": "sinusoid", "enabled": True, "config": {}}
+        post_url = "/foglamp/service"
+        conn = http.client.HTTPConnection(foglamp_url)
+        conn.request("POST", post_url, json.dumps(payload))
+        res = conn.getresponse()
+        assert 200 == res.status, "ERROR! POST {} request failed".format(post_url)
+
+        copy_certs()
+
+        payload = {"name": task_name,
+                   "plugin": "{}".format(north_plugin),
+                   "type": "north",
+                   "schedule_type": 3,
+                   "schedule_repeat": 5,
+                   "schedule_enabled": True,
+                   "config": {"project_id": {"value": "foglamp"},
```

Comment:
We should probably have some notes at least as to the GCP setup we are assuming and the certs etc.

Reply:
I have added this as [prerequisite](https://github.com/foglamp/FogLAMP/pull/1804/files#diff-2cae579d734ab10093cc71381e01aff2R9).